### PR TITLE
Add prefix sum solution

### PR DESCRIPTION
**Pull request by 96f4**
This problem is in More on Prefix Sums

_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making the changes.

### CODE REVIEW

File: solutions/silver/usaco-416.mdx [hunk 1]
```diff
@@ -7,8 +7,94 @@ author: Brad Ma, Kevin Sheng
 
 [Official Analysis (C++)](http://www.usaco.org/current/data/sol_lazy_silver.html)
 
-## Implementation
+### Implementation (Prefix Sums)
```
Comment (SansPapyrus683) [inline solutions/silver/usaco-416.mdx:10]: put this below the previous sol

File: solutions/silver/usaco-416.mdx [hunk 2]
```diff
@@ -7,8 +7,94 @@ author: Brad Ma, Kevin Sheng
...
+for i in range(new_n):
+	for j in range(new_n):
+		prefix[i + 1][j + 1] = field[i][j] + prefix[i + 1][j] + prefix[i][j + 1] - prefix[i][j]
+
+most_grass = max(
+	(prefix[i + k + 1][j + k + 1] - prefix[i + k + 1][j - k] - prefix[i - k][j + k + 1] + prefix[i - k][j - k]
+	for i in range(k, new_n - k) for j in range(k, new_n - k)),
+	# if the generator is empty, k >= n, so Bessie can reach everything
+	default=prefix[new_n][new_n]
+)
```
Comment (SansPapyrus683) [inline solutions/silver/usaco-416.mdx:90]: at this point just use a for loop

File: solutions/silver/usaco-416.mdx [hunk 3]
```diff
@@ -7,8 +7,94 @@ author: Brad Ma, Kevin Sheng
...
+	vector<vector<int>> prefix(new_n + 1, vector<int>(new_n + 1));
+	for (int i = 0; i < new_n; i++) {
+		for (int j = 0; j < new_n; j++) {
+			prefix[i + 1][j + 1] = field[i][j] + prefix[i + 1][j] + prefix[i][j + 1] - prefix[i][j];
+		}
+	}
+	int most_grass = 0;
+	for (int i = k; i < new_n - k; i++) {
+		for (int j = k; j < new_n - k; j++) {
+			most_grass = max(most_grass, prefix[i + k + 1][j + k + 1] - prefix[i + k + 1][j - k] - prefix[i - k][j + k + 1] + prefix[i - k][j - k]);
```
Comment (SansPapyrus683) [inline solutions/silver/usaco-416.mdx:51]: over 80 chars

File: solutions/silver/usaco-416.mdx [hunk 4]
```diff
@@ -7,8 +7,94 @@ author: Brad Ma, Kevin Sheng
...
+
+int main() {
+	std::ifstream read("lazy.in");
+
+	int n;
+	int k;
+	read >> n >> k;
+	// the side length needed to accomodate the 45 degree rotation
+	int new_n = 2 * n - 1;
+	vector<vector<int>> field(new_n, vector<int>(new_n));
```
Comment (SansPapyrus683) [inline solutions/silver/usaco-416.mdx:34]: ngl don't think this is sufficiently different from the og sol to warrant a diff section?
by "prefix sums" i was thinking storing pref sums for every row and iterating through them for each query

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
start slow

File: solutions/silver/usaco-416.mdx (hunk 4 above)
Comment (SansPapyrus683) [reply solutions/silver/usaco-416.mdx]: elaborate?

File: solutions/silver/usaco-416.mdx (hunk 2 above)
Comment (SansPapyrus683) [reply solutions/silver/usaco-416.mdx]: can i have a screenie of it tle-ing on the last case? just wanna make sure

Comment (SansPapyrus683) [reply solutions/silver/usaco-416.mdx]: i'd keep the for loop & add a warning like "may tle on last tc"

File: solutions/silver/usaco-416.mdx [hunk 5]
```diff
@@ -145,3 +145,110 @@ public class Lazy {
...
+				most_grass, 
+				prefix[i + k + 1][j + k + 1]
+				- prefix[i + k + 1][j - k]
+				- prefix[i - k][j + k + 1]
+				+ prefix[i - k][j - k]
+			);
+		}
+	}
+	if (k >= n) // the loop above did not run, and Bessie can reach everything
+		most_grass = prefix[new_n][new_n];
```
Comment (SansPapyrus683) [inline solutions/silver/usaco-416.mdx:200]: braces

File: solutions/silver/usaco-416.mdx [hunk 6]
```diff
@@ -145,3 +145,110 @@ public class Lazy {
...
+	// read input and store it rotated by 45 degrees
+	for (int i = 0; i < n; i++) {
+		for (int j = 0; j < n; j++) {
+			read >> field[i + j][n - i + j - 1];
+		}
+	}
+	vector<vector<int>> prefix(new_n + 1, vector<int>(new_n + 1));
+	for (int i = 0; i < new_n; i++) {
+		for (int j = 0; j < new_n; j++) {
+			prefix[i + 1][j + 1] = field[i][j] + prefix[i + 1][j] + prefix[i][j + 1] - prefix[i][j];
```
Comment (SansPapyrus683) [inline solutions/silver/usaco-416.mdx:184]: over 80 chars

File: solutions/silver/usaco-416.mdx [hunk 7]
```diff
@@ -145,3 +145,110 @@ public class Lazy {
...
+
+# read input and store it rotated by 45 degrees
+for i in range(n):
+	for j, x in enumerate(map(int, input().split())):
+		field[i + j][n - i + j - 1] = x
+
+prefix = [[0] * (new_n + 1) for _ in range(new_n + 1)]
+for i in range(new_n):
+	for j in range(new_n):
+		prefix[i + 1][j + 1] = field[i][j] + prefix[i + 1][j] + prefix[i][j + 1] - prefix[i][j]
```
Comment (SansPapyrus683) [inline solutions/silver/usaco-416.mdx:234]: likewise

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

**Review by SansPapyrus683 — APPROVED** [review]
@envyaims @ryanchou-dev get in here

**Review by envyaims — APPROVED** [review]
your code looks exactly like something sans would write 😒

**Comment by SansPapyrus683** [comment]
cope and seethe envy

**Comment by ryanchou-dev** [comment]
bro i was js reviewing grrr

**Comment by SansPapyrus683** [comment]
sorry ryan!

**Pull request merged**